fmt/printf.h

```diff
@@ -588,6 +600,8 @@ FMT_VARIADIC(int, printf, CStringRef)
   \endrst
  */
 inline int fprintf(std::ostream &os, CStringRef format_str, ArgList args) {
+  std::mutex mtx;
+  std::unique_lock<std::mutex> lck(mtx);
```

Comment:
Different school of thought, I think thread safety should be provided by the library in 2017, it could be implemented differently and more efficiently. I have one last question is it a thing of not wanting to add thread safety or a matter of implementation?